jwst/skymatch/skymatch_step.py

```diff
@@ -206,30 +176,23 @@ def _imodel2skyim(self, image_model):
             pix_area=1.0,  # TODO: pixel area
             convf=1.0,  # TODO: conv. factor to brightness
             mask=dqmask,
-            id=image_model.meta.filename,  # file name?
+            id=image_model.meta.filename,
             skystat=self._skystat,
             stepsize=self.stepsize,
-            reduce_memory_usage=self._is_asn,
-            meta={'image_model': input_image_model}
+            reduce_memory_usage=False,  # this overwrote input files
+            meta={'index': index}
         )
 
-        if self._is_asn:
-            image_model.close()
-
         if self.subtract:
             sky_im.sky = level
 
         return sky_im
 
-    def _set_sky_background(self, sky_image, step_status):
-        image = sky_image.meta['image_model']
+    def _set_sky_background(self, sky_image, library, step_status):
```

Comment:
Maybe a note here in the docstring that this must be called with an open library.